client/src/editor/filters/edge-glow.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from "react";
+import { BaseFilter } from "./base-filter";
+// Utility function to create neon colors for edges
+
+function getColorFromPalette(
+  value: number,
+  palette: string
+): [number, number, number] {
+  switch (palette) {
+    case "psychedelic":
+      return [
+        Math.sin(value * 0.1) * 127 + 128,
+        Math.sin(value * 0.2 + 2) * 127 + 128,
+        Math.sin(value * 0.3 + 4) * 127 + 128,
+      ];
+    case "cool":
+      return [
+        0,
+        Math.sin(value * 0.3) * 127 + 128,
+        Math.cos(value * 0.2) * 127 + 128,
+      ];
+    case "warm":
+      return [
+        Math.cos(value * 0.3) * 127 + 128,
+        Math.sin(value * 0.2) * 50 + 100,
+        0,
+      ];
+    case "retro":
+      return [
+        Math.cos(value * 0.5) * 127 + 200,
+        Math.sin(value * 0.4) * 127 + 200,
+        Math.cos(value * 0.3) * 127 + 200,
+      ];
+    case "firestorm":
+      return [
+        Math.sin(value * 0.3) * 255,
+        Math.sin(value * 0.2) * 127 + 128,
+        0,
+      ];
+    default:
+      return [255, 255, 255];
+  }
+}
+// Sobel filter implementation
+function applySobelFilter(image: ImageData, palette: string): ImageData {
+  const { width, height, data } = image;
+  const output = new ImageData(width, height);
+  const sobelX = [-1, 0, 1, -2, 0, 2, -1, 0, 1];
+  const sobelY = [-1, -2, -1, 0, 0, 0, 1, 2, 1];
+
+  const getPixel = (x: number, y: number, c: number) => {
+    const idx = (y * width + x) * 4 + c;
+    return data[idx] || 0;
+  };
+
+  for (let y = 1; y < height - 1; y++) {
+    for (let x = 1; x < width - 1; x++) {
+      let gx = 0,
+        gy = 0;
+
+      for (let ky = -1; ky <= 1; ky++) {
+        for (let kx = -1; kx <= 1; kx++) {
+          const weightX = sobelX[(ky + 1) * 3 + (kx + 1)];
+          const weightY = sobelY[(ky + 1) * 3 + (kx + 1)];
+          const intensity =
+            (getPixel(x + kx, y + ky, 0) +
+              getPixel(x + kx, y + ky, 1) +
+              getPixel(x + kx, y + ky, 2)) /
+            3;
+
+          gx += weightX * intensity;
+          gy += weightY * intensity;
+        }
+      }
+
+      const edgeMagnitude = Math.sqrt(gx ** 2 + gy ** 2);
+      const [r, g, b] = getColorFromPalette(edgeMagnitude, palette);
+
+      const idx = (y * width + x) * 4;
+      output.data[idx] = r;
+      output.data[idx + 1] = g;
+      output.data[idx + 2] = b;
+      output.data[idx + 3] = 255; // Alpha channel
+    }
+  }
```

Comment:
Later we will need to move this function to filtering-worker.ts. But its OK for now, we will do that in a separate PR